### PR TITLE
Add option for `reuse_port` in bind

### PR DESCRIPTION
**Pull request by yoelbassin**
Hey!
I use this lib in one of my projects, and I need multiple socket instances to bind to the same (IP, port) address.

asyncio's `loop.create_datagram_endpoint` supports the argument `reuse_port` to allow the endpoint to be bound to the same port as other existing endpoints are bound to, so long as they all set this flag when being created (however, this option isn't supported on Windows and some Unixes).

I added this argument to the bind function along with tests for this feature.

Thank you for your great work!

### CODE REVIEW

**Comment by yoelbassin** [comment]
This option is only relevant to python > 3.5, since there, the `reuse_address` argument was still valid, and if not specified it was automatically set to True on UNIX.
Since python >= 3.6.10: the `reuse_address` parameter is no longer supported due to security concerns.

File: test/test_aio.py [hunk 1]
```diff
@@ -492,3 +492,24 @@ async def test_transport_closed():
...
+async def test_bind_reuse_port():
+    async def use_socket(addr, reuse_port=None):
+        sock = await asyncio_dgram.bind(addr, reuse_port=reuse_port)
+        # give gather time to move to the other uses after the bind
+        await asyncio.sleep(0.1)
+        sock.close()
+
+    addr = ("127.0.0.1", 53001)
+    clients_count = 10
+    with pytest.raises(OSError):
```
Comment (jsbronder) [inline test/test_aio.py:509]: I'd add a `match` here as there's a lot of potential things that could raise OSError.

`with pytest.raises(OSError, match="Address already in use")`

File: asyncio_dgram/aio.py [hunk 2]
```diff
@@ -213,15 +213,20 @@ def resume_writing(self):
         super().resume_writing()
 
 
-async def bind(addr):
+async def bind(addr, reuse_port=None):
```
Comment (jsbronder) [inline asyncio_dgram/aio.py:216]: Please squash this commit (and the next one) into the previous commit.  I want to be sure that every commit in `master` will pass `make format lint test`.  This enables scripting git-bisect more easily.

**Review by jsbronder — CHANGES_REQUESTED** [review]
Please address the comments I left and the lint error from CI and I'll merge.

(Force pushing after you squash commits is perfectly fine.)

**Pull request merged**